File: src/main/java/com/ggardet/patient/spi/MedicationService.java
package com.ggardet.patient.spi;

import com.ggardet.patient.spi.dto.MedicationDTO;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface MedicationService {
    MedicationDTO create(final MedicationDTO medicationDTO);
    MedicationDTO findById(final UUID id);
    List<MedicationDTO> findAll();
    List<MedicationDTO> findByPatientId(final UUID patientId);
    List<MedicationDTO> findByPrescribingDoctorId(final UUID doctorId);
    List<MedicationDTO> findActiveMedications();
    List<MedicationDTO> findActiveMedicationsAfterDate(final LocalDate date);
    MedicationDTO update(final UUID id, final MedicationDTO medicationDTO);
    void deleteById(final UUID id);
}

File: src/main/kotlin/com/ggardet/patient/spi/MedicationService.kt
package com.ggardet.patient.spi

import com.ggardet.patient.spi.dto.MedicationDTO
import java.time.LocalDate
import java.util.*

interface MedicationService {
    fun create(medicationDTO: MedicationDTO): MedicationDTO
    fun findById(id: UUID): MedicationDTO
    fun findAll(): List<MedicationDTO>
    fun findByPatientId(patientId: UUID): List<MedicationDTO>
    fun findByPrescribingDoctorId(doctorId: UUID): List<MedicationDTO>
    fun findActiveMedications(): List<MedicationDTO>
    fun findActiveMedicationsAfterDate(date: LocalDate): List<MedicationDTO>
    fun update(id: UUID, medicationDTO: MedicationDTO): MedicationDTO
    fun deleteById(id: UUID)
}